File: app/src/main/java/com/th/forge/taxiorders/OrderDetailActivity.java

package com.th.forge.taxiorders;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;

import com.th.forge.taxiorders.entity.Order;

public class OrderDetailActivity extends SingleFragmentActivity implements OrderDetailFragment.OnFragmentInteractionListener {

    private static final String ORDER_KEY = "order_key";

    public static Intent newIntent(Context context, Order order) {
        Intent i = new Intent(context, OrderDetailActivity.class);
        Bundle bundle = new Bundle();
        bundle.putSerializable(ORDER_KEY, order);
        i.putExtras(bundle);
        return i;
    }

    @Override
    protected Fragment createFragment() {
        return OrderDetailFragment.newInstance(getIntent().getSerializableExtra(ORDER_KEY));
    }

    @Override
    public void onFragmentInteraction(String title, String subtitle) {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(title);
            actionBar.setSubtitle(subtitle);
        }
    }
}

File: app/src/main/java/com/th/forge/taxiorders/OrderDetailActivity.kt
package com.th.forge.taxiorders

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.ActionBar

import com.th.forge.taxiorders.entity.Order

class OrderDetailActivity : SingleFragmentActivity(), OrderDetailFragment.OnFragmentInteractionListener {

    override fun createFragment(): Fragment {
        return OrderDetailFragment.newInstance(intent.getSerializableExtra(ORDER_KEY))
    }

    override fun onFragmentInteraction(title: String, subtitle: String) {
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = title
            actionBar.subtitle = subtitle
        }
    }

    companion object {

        private val ORDER_KEY = "order_key"

        fun newIntent(context: Context, order: Order): Intent {
            val i = Intent(context, OrderDetailActivity::class.java)
            val bundle = Bundle()
            bundle.putSerializable(ORDER_KEY, order)
            i.putExtras(bundle)
            return i
        }
    }
}
